data/org.freedesktop.portal.Notification.xml

```diff
@@ -106,21 +180,361 @@
 
           * ``label`` (``s``)
 
-            User-visible label for the button. Mandatory.
+            User-visible label for the button.
+            If this property is missing the action may not be displayed.
 
           * ``action`` (``s``)
 
             Name of an action that is exported by the application. The action
-            will be activated when the user clicks on the button. Mandatory.
+            will be activated when the user clicks on the button.
+            If this property is missing the action may not be displayed.
+
+          * ``target`` (``v``)
+
+            Target parameter to send along when activating the action.
+
+          * ``icon`` (``v``)
+
+            A serialized icon to be added to a button. The format for
+            is the same as for the notification ``icon``.
+
+          * ``purpose`` (``s``)
+
+            The ``purpose`` of the button. This information may be used by the
+            notification server to style the button specially.
+
+            The standardized hints are defined as part of a ``content-type``.
+            Applications are allowed to use any string as a ``purpose`` but
+            should stick to the form ``class.specific``.
+            If the server doesn't understand the ``purpose` it will be
+            ignored and the button will be shown as a normal button.
+
+            Additional purposes may be defined by notification servers.
+
+          Any of the above button properties may be omitted, if the notification server
+          can't display the button with the given information it won't display it at all.
```

Comment:
Just to double-check: An ongoing call notification could add an action to hold the call w/ an action and target set, but no label or icon set, so that the notification server knows it's possible to place the call on hold but doesn't ever render the button. Right?

Reply:
Ah, never mind. I see `actions` below